webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/web/services/SecretService.scala

```diff
@@ -0,0 +1,212 @@
+/*
+*************************************************************************************
+* Copyright 2021 Normation SAS
+*************************************************************************************
+*
+* This file is part of Rudder.
+*
+* Rudder is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* In accordance with the terms of section 7 (7. Additional Terms.) of
+* the GNU General Public License version 3, the copyright holders add
+* the following Additional permissions:
+* Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+* Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+* Public License version 3, when you create a Related Module, this
+* Related Module is not considered as a part of the work and may be
+* distributed under the license agreement of your choice.
+* A "Related Module" means a set of sources files including their
+* documentation that, without modification of the Source Code, enables
+* supplementary functions or services in addition to those offered by
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+package com.normation.rudder.web.services
+
+import better.files.File
+import com.normation.NamedZioLogger
+import com.normation.errors.IOResult
+import com.normation.errors.Inconsistency
+import com.normation.eventlog.ModificationId
+import com.normation.rudder.domain.secrets._
+import com.normation.rudder.repository.EventLogRepository
+import com.normation.rudder.web.services.CurrentUserService.actor
+import com.normation.utils.StringUuidGenerator
+import net.liftweb.json.JsonAST.JValue
+import net.liftweb.json.JsonDSL._
+import net.liftweb.json.JString
+import net.liftweb.json.JsonParser
+import zio._
+import zio.syntax._
+
+import java.nio.charset.StandardCharsets
+
+trait SecretService {
+  def getAllSecret: IOResult[List[Secret]]
+  def getSecretById(secretId: String): IOResult[Option[Secret]]
+  def addSecret(s: Secret, reason: String): IOResult[Unit]
+  def deleteSecret(secretId: String, reason: String): IOResult[Unit]
+  def updateSecret(newSecret: Secret, reason: String): IOResult[Unit]
+}
+
+class FileSystemSecretRepository(
+    jsonDbPath   : String
```

Comment:
I think it should be configuration-repository path, and the end of the path should be in the repository 

btw, the secrets.json should be in a secrets directory